src/common-types/middleware-locals.d.ts

```diff
@@ -0,0 +1,20 @@
+type CommonResponseLocals = {
+	nonce: string
+}
+
+type JiraHostVerifiedLocals = CommonResponseLocals  & {
+	jiraHost: string,
+};
+
+type JiraJwtVerifiedLocals = CommonResponseLocals & {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	installation: any //TODO: This is a hack, fix it with full definition.
+};
+
+type GitHubAppVerifiedLocals = CommonResponseLocals & {
+	gitHubAppConfig: GitHubAppConfig,
+};
+
+type GitHubUserTokenVerifiedLocals = CommonResponseLocals & {
+	githubToken: string;
+};
```

Comment:
Wouldn't it be easier to just have a single object with all locals, all of which are optional?

Reply:
So this is based on the ***`Assumption: routing is setup properly and securely.`*** 
With above assumption, handlers can mix and use different `XXXVerifiedLocals`, and having a `non-optional` field in it (because of the assumption).

We do have same assumption right now in prod just not typed it (see my other comments) bellow.

---

As how to address/impl the assumption itself, it can be another PR.